File: pset6/caesar.py
import sys
from sys import argv
from cs50 import get_string

if len(argv) != 2:
    print("Invalid input")
    sys.exit(1)

k = int(argv[1])

plainText = get_string("Plain text: ")
print("ciphertext: ", end="")

#find length of plaintext
lengthPlainText = len(plainText)

#iterate through plaintext letters
for i in range(lengthPlainText):

    #check if lowercase
    if ord(plainText[i]) >= (ord('a')) and ord(plainText[i]) <= (ord('z')):
        #shift letter using k and % 26
        cipher = ((((ord(plainText[i]) - (ord('a'))) + k) % 26) + (ord('a')))
        print(chr(cipher),end="")

    #check if upper case
    elif ord(plainText[i]) >= (ord('A')) and ord(plainText[i]) <= (ord('Z')):
        #shift letter using k and % 26
        cipher = ((((ord(plainText[i]) - (ord('A'))) + k) % 26) + (ord('A')))
        print(chr(cipher), end="")
    #else, print the character
    else:
        print(plainText[i], end="")

print()

File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <math.h>


int main(int argc, string argv[])
{

if (argc != 2)
    {
      //If there are an incorrect number of arguments:
         printf("Usage: ./caesar key\n");
         return 1;

    }
    
    int k = atoi(argv[1]); //Convert into an integer

    string plainText = get_string("plaintext: ");
    printf("ciphertext: "); //title for the cipher text

    int n = strlen(plainText);
    //iterate through text letter by letter
    for (int i = 0; i < n; i++) {

       if (plainText[i] >= 'a' && plainText[i] <= 'z')
       {
           //Check if is lowercase
           printf("%c", (((plainText[i] - 'a') + k) % 26) + 'a');// print out lowercase with key
       } else if (plainText[i] >= 'A' && plainText[i] <= 'Z')

           {
               printf("%c", (((plainText[i] - 'A') + k) % 26) + 'A');
           } else

           {
               printf("%c", plainText[i]);
           }
       }
    printf("\n");
    return 0;
}

   int n = strlen(argv[1]);

    //iterate through text letter by letter
    for (int i = 0; i < n; i++)
    {
        printf("%c", isalpha(i));
    }

    string plainText = get_string("plaintext: ");
    printf("ciphertext: "); //title for the cipher text

    int length = strlen(plainText);
    //iterate through text letter by letter
    for (int i = 0; i < length; i++) {

       if (plainText[i] >= 'a' && plainText[i] <= 'z')
       {
           //Check if is lowercase
           printf("%c", (((plainText[i] - 'a') + k) % 26) + 'a');// print out lowercase with key
       } else if (plainText[i] >= 'A' && plainText[i] <= 'Z')

           {
               printf("%c", (((plainText[i] - 'A') + k) % 26) + 'A');
           } else

           {
               printf("%c", plainText[i]);
           }
       }
    printf("\n");
    return 0;
}
